### PR TITLE
Split subjectChoice for different usage and rename to SchoolSubject

### PR DESCRIPTION
**Pull request by mcginkel**
Split subject choice in student subject choice with dates.
Changed to capitals in model to get consistency.

### CODE REVIEW

File: reference/sisdata.v1.yaml [hunk 1]
```diff
@@ -325,11 +360,11 @@ components:
         startDate:
           type: string
           format: date
-          description: Start of this subject for the student. Date cannot be in the future.
+          description: Start of this subject for the student. (ISO8601)
         endDate:
           type: string
           format: date
-          description: Start of this subject for the student. Date cannot be in the future. null if not ended
+          description: Start of this subject for the student. (ISO8601). null if not ended
```
Comment (Jorimvdw) [inline reference/sisdata.v1.yaml:367]: * End of this subject for the student

Comment (Jorimvdw) [reply reference/sisdata.v1.yaml:367]: Wat bedoelen we hier eigenlijk met _null if not ended?_ null if no enddate is given?

File: reference/sisdata.v1.yaml [hunk 2]
```diff
@@ -278,27 +278,27 @@ components:
...
         teachers:
           type: array
           items:
-            $ref: '#/components/schemas/teacherReference'
+            $ref: '#/components/schemas/TeacherReference'
         subjectChoices:
           type: array
           items:
-            $ref: '#/components/schemas/subject-choice'
+            $ref: '#/components/schemas/SubjectChoice'
```
Comment (Jorimvdw) [inline reference/sisdata.v1.yaml:293]: Ik zou zelf gaan voor SchoolSubject, dan ontstaat er zeker geen spraakverwarring met de StudentSubjectChoice

File: reference/sisdata.v1.yaml (hunk 1 above)
Comment (mcginkel) [reply reference/sisdata.v1.yaml:367]: null if you do not know the end date, so it is until end of schoolyear?

File: reference/sisdata.v1.yaml (hunk 2 above)
Comment (mcginkel) [reply reference/sisdata.v1.yaml:293]: I think I will do another proposal: split the student specific fields (start/end), so SubjectChoice is always the same

Comment (Jorimvdw) [reply reference/sisdata.v1.yaml:293]: Or just exchange only a ref to the subject with a start- and enddate. So you can get all the SchoolSubjects first, and with the Student-object you just reference a SchoolSubject

**Comment by mcginkel** [comment]
Changed scope of the PR after tech meeting. we will change the name SubjectChoice to SchoolSubject and make sure we only have one version/definition. If a students needs start/end dat then these are modelled seperately.

**Pull request merged**